src/net.cpp

```diff
@@ -1308,6 +1308,7 @@ void CConnman::GenerateSelectSet(std::set<SOCKET> &recv_set, std::set<SOCKET> &s
     }
 }
 
+#ifdef WIN32
```

Comment:
Apparently [`poll` has been flaky on previous versions of MacOS](https://daniel.haxx.se/blog/2016/10/11/poll-on-mac-10-12-is-broken/). Do we want to fall back to `select` on Darwin systems just to be safe?

Reply:
I'm avoiding using poll() or select() to sleep specifically because of this class of bug.